accepted/future-releases/constructor-tearoffs/feature-specification.md

```diff
@@ -402,6 +402,18 @@ A compile-time error occurs if a constructor tear-off denotes a generative const
 
 *A generative constructor declared in an abstract class may well be executed as part of the execution of a constructor tear-off, via a superinitializer. It cannot, however, be torn off on its own, because the execution of a constructor tear-off is an instance creation, and abstract classes do not have instances. There is no similar constraint on a factory constructor.*
 
+### Compile-time errors
+
+This section lists the compile-time errors associated with this feature. The _italic_ entries are commentary, which indicates that they are stated elsewhere already.
+
+It is compile-time error if an expression of the form `e.new <parameterPart>?` occurs, and `e` is not a type literal.
```

Comment:
Can you can drop the `<parameterPart>?`? Since it's optional, it's an error without it anyway, so why mention it at all?
If `e.new` occurs *as an expression* and *e* is not a type literal, then it doesn't matter what it's followed by.

Reply:
I don't think that would work (to drop `<parameterPart>?`). We generally specify the rules for certain syntactic forms, and there is no reason to assume that a short sequence _s_ of tokens has any properties that are valid for any sequence _s1_ of tokens which is obtained from _s_ by adding extra tokens at the end. In particular, just because we've said that `e.new` is an error when `e` satisfies certain criteria, we haven't said anything at all about the error/non-error status of something which is derived from `e.new <parameterPart>`.

The reason why we can talk about the latter at all is that we deal with the properties of such constructs (like an instance creation expression) as a whole. The rules about an `e.new` which is followed by something else are simply irrelevant for `e.new <parameterPart>`.

We might require that we specify this more explicitly, such that we talk about `e.new`-which-is-not-followed-by-`<parameterPart>` vs. `e.new <parameterPart>`, but we don't do that consistently, and I think it would give rise to a very verbose style.

So I did not change anything here at this point.